src/DIRAC/Resources/Catalog/RucioFileCatalogClient.py

```diff
@@ -12,24 +12,30 @@
 
 import DIRAC
 from DIRAC import S_OK, S_ERROR, gLogger
+from DIRAC.Core.Security import Locations
 from DIRAC.Resources.Catalog.Utilities import checkCatalogArguments
 from DIRAC.Core.Utilities.List import breakListIntoChunks
-from DIRAC.Core.Security.ProxyInfo import getProxyInfo
+from DIRAC.Core.Security.ProxyInfo import getProxyInfo, getVOfromProxyGroup
 from DIRAC.Resources.Catalog.FileCatalogClientBase import FileCatalogClientBase
 
 from rucio.client import Client
 from rucio.common.exception import DataIdentifierNotFound, FileReplicaAlreadyExists, \
     CannotAuthenticate, MissingClientParameter
 from rucio.common.utils import chunks, extract_scope
 sLog = gLogger.getSubLogger(__name__)
+useDiracScopeAlg = False
 
 
 def get_scope(lfn, scopes=None):
-  """Helper function that extract the scope from the LFN
+  """Helper function that extracts the scope from the LFN
   """
-  if scopes is None:
-    scopes = []
-  scope, _ = extract_scope(did=lfn, scopes=scopes)
+  #  default Dirac scope algorithm:
+  if useDiracScopeAlg:
+    scope = lfn.split('/')[1]
```

Comment:
No, that won't work. The `add_file` method defined on the Rucio server side needs to use the same method to extract the scope as the one used by the client. This is the server code:
https://github.com/rucio/rucio/blob/master/lib/rucio/core/dirac.py#L74

Reply:
You are right here. In one of my previous attempts I used Dirac add file to add a file, not the Rucio server code. Could this be somehow implemented in parallel ? I mean, for non-config file people we use dirac upload mechanism, not dirac.py from Rucio. I'll look into this. More than that. Your scope algorithm is a part of Rucio, the only one along with the default one. Is it a way to go, that every new collaboration willing to use Rucio will have to put their algorithm into Rucio code ?

Reply:
Or more precisely upload a file and register it in Rucio on the client side, not the server side. In this case Rucio will know the scope and can do all replication work. The scope algorithm would not be needed by the server anymore. Am I right ?

Reply:
After having looked into this again, I think it is necessary that `RucioFileCatalogClient` inherits from diracclient:
https://github.com/rucio/rucio/blob/b6092bcd78f8cfac079e5b3cc25a92b4163960ed/lib/rucio/client/diracclient.py#L32

and not from the base client, as it does now. The scope issue still remains.

Reply:
Is it any reason we cannot modify `add_files `on a client to take a list of DIDs instead of a list of LFNs ? In this case we could eliminate a call to `extract_scope` on the server side. We would need a modified version of it on the client side but is this case we could get a name of the algorithm from the CS (multi-VO issue is solved by this as well), thus avoiding a config file access.

Reply:
> After having looked into this again, I think it is necessary that RucioFileCatalogClient inherits from diracclient:
No, it needs to inherit from `baseclient` because some other methods (e.g. listDirectory) use methods from it.
 > Is it any reason we cannot modify add_files on a client to take a list of DIDs instead of a list of LFNs ?
 Yes, the main reason is that when `add_files` is called, it is supposed to create the ancestors (parents, grand-parents, etc.) if they do not exists. If you pass only the scope of the file, Rucio won't know which scope to use for the parents.

Reply:
Sorry, I messed up a bit in my first question. Everything is fine here. 

Concerning the second question I think I see your point. The scope is extracted more that once on the server.
The algorithm is in the _utils,_ it is needed on the server. Fine. There is a config file on the server - no issue here.

How would we give the client the right scope which uses the algorithm X the name of which the client takes from the CS ?

Ask the server ?? to extract the scope ? This would be one more remote call at the client en lieu of the current `extract_scope`. Feasible ?

Reply:
I would really avoid to call the server on each LFN to extract the scope